# Everyone Can Learn Data Scholarship

## Understanding Our Movie Data 🎥 


🎥 Challenge 2: You have just been hired by a large movie studio to perform data analysis. Your manager, an executive at the company, wants to make new movies that 'recapture the magic of old Hollywood.' So you've decided to look at the most successful films to help generate ideas that could turn into future successful films.

In [26]:
SELECT * 
FROM cinema.films
LIMIT 10

id                                             title  ...      gross     budget
0   1  Intolerance: Love's Struggle Throughout the Ages  ...        NaN   385907.0
1   2                    Over the Hill to the Poorhouse  ...  3000000.0   100000.0
2   3                                    The Big Parade  ...        NaN   245000.0
3   4                                        Metropolis  ...    26435.0  6000000.0
4   5                                     Pandora's Box  ...     9950.0        NaN
5   6                               The Broadway Melody  ...  2808000.0   379000.0
6   7                                     Hell's Angels  ...        NaN  3950000.0
7   8                                A Farewell to Arms  ...        NaN   800000.0
8   9                                       42nd Street  ...  2300000.0   439000.0
9  10                                She Done Him Wrong  ...        NaN   200000.0

[10 rows x 9 columns]

**0. Let us verify if there are any movies produced after 1997 included in the dataset to ensure that we perform our analysis on the rows involved in our context of study.**

In [27]:
SELECT COUNT(*) 
FROM cinema.films
WHERE release_year > 1997;

count
0   3891

In [28]:
SELECT COUNT(*) 
FROM cinema.films

count
0   4968

There are 3,891 out of 4,968 movies that were released after 1997, indicating that our dataset is broader than necessary. For the purpose of our study, we will query our data for movies that fall into the categories of "Movies Released Before 1997" and "general scope."

### 1. How many movies are present in the database?

To get the count of movies, we are going to count as per their IDs and their titles as well, to not only get the number of movies in the database but verify if there are any duplicates that could render our analysis faulty

**General Scope**

In [29]:
-- Selects the count of distinct movie titles from the cinema.films table
SELECT COUNT(DISTINCT title) AS Movie_Count_Title
FROM cinema.films

movie_count_title
0               4844

In [30]:
-- Select the count of distinct film IDs from the cinema.films table
SELECT COUNT(DISTINCT id) AS Movie_Count_id
FROM cinema.films

movie_count_id
0            4968

**Movies Released Before 1997**

In [31]:
-- Select the count of distinct movie titles
SELECT COUNT(DISTINCT title) AS Movie_Count_Title
-- From the table cinema.films
FROM cinema.films
-- Where the release year is before 1997
WHERE release_year < 1997

movie_count_title
0                893

In [32]:
-- Select the count of distinct movie IDs from the cinema.films table
SELECT COUNT (DISTINCT id) AS Movie_Count_id
-- Filter the movies to include only those released before the year 1997
FROM cinema.films
WHERE release_year < 1997;

movie_count_id
0             917

We observe that when counting the movies by their ID,there is a higher count compared to counting by their title or name. This discrepancy may suggest different issues such as : different movies sharing the same names, a single movie being translated into multiple languages or maybe there are simply duplicate entries.

If a movie simply has the same title as another movie, but is completely unrelated, then they will most likely have a different release_year and/or duration. If it is a movie which has simply been translated, then the release year and duration won't differ. Only the language should, upon our assumption. 

Therefore we will consider as duplicates, movies that have the same title, release year and duration

Let us verify duplicate entries to ensure we will be counting the exact number of movies in our database

In [33]:
-- Selects the id, title, release_year, country, duration, language, certification, gross, and budget
-- from the cinema.films table where the combination of title, release_year, country, and language
-- appears more than once in the table.

SELECT id, title, release_year, country, duration, language, certification, gross, budget
FROM cinema.films
WHERE (title, release_year,duration) IN (
    -- Subquery to find combinations of title, release_year, duration
    -- that appear more than once in the cinema.films table.
    SELECT title, release_year, duration
    FROM cinema.films
    GROUP BY title, release_year,duration
    HAVING COUNT(*) > 1
);

id                         title  ...        gross       budget
0      59  20,000 Leagues Under the Sea  ...          NaN    5000000.0
1      60  20,000 Leagues Under the Sea  ...          NaN    5000000.0
2      85                        Lolita  ...          NaN    2000000.0
3      86                        Lolita  ...          NaN    2000000.0
4     130      Night of the Living Dead  ...          NaN     114000.0
..    ...                           ...  ...          ...          ...
232  4855           Godzilla Resurgence  ...          NaN          NaN
233  4906               The Jungle Book  ...  362645141.0  175000000.0
234  4907               The Jungle Book  ...  362645141.0  175000000.0
235  4908          The Legend of Tarzan  ...  124051759.0  180000000.0
236  4909          The Legend of Tarzan  ...  124051759.0  180000000.0

[237 rows x 9 columns]

As suspected, there are indeed multiple movies repeating themselves. There are 237 unique movies along with their duplicates in total.

To obtain the actual count of distinct movies, let us query our database and include a clause to handle these duplicates.

In [34]:
SELECT COUNT(*)
FROM (
    SELECT DISTINCT title, release_year, language
    FROM cinema.films
) AS distinct_films;

count
0   4846

**Answer** : There are 4846 movies

The scope of our analysis mainly involves movies released before 1997. To get a better idea of our data, let us see how many duplicates are involved in our scope

**Movies Released Before 1997**

In [35]:
SELECT id, title, release_year, country, duration, language, certification, gross, budget
FROM cinema.films
WHERE (title, release_year, country, language) IN (
    SELECT title, release_year, country, language
    FROM cinema.films
	WHERE cinema.films.release_year < 1997
    GROUP BY title, release_year, country, language
    HAVING COUNT(*) > 1
);


id                          title  ...        gross    budget
0    59   20,000 Leagues Under the Sea  ...          NaN   5000000
1    60   20,000 Leagues Under the Sea  ...          NaN   5000000
2    85                         Lolita  ...          NaN   2000000
3    86                         Lolita  ...          NaN   2000000
4   130       Night of the Living Dead  ...          NaN    114000
5   131       Night of the Living Dead  ...          NaN    114000
6   166          The French Connection  ...          NaN   1800000
7   167          The French Connection  ...          NaN   1800000
8   194   The Texas Chain Saw Massacre  ...   30859000.0     83532
9   195   The Texas Chain Saw Massacre  ...   30859000.0     83532
10  211                       The Omen  ...          NaN   2800000
11  212                       The Omen  ...          NaN   2800000
12  236                      Halloween  ...   47000000.0    300000
13  237                      Halloween  ...   47000000.0    300000
14  238                      Halloween  ...   47000000.0    300000
15  281                        The Fog  ...   21378000.0   1000000
16  282                        The Fog  ...   21378000.0   1000000
17  295   History of the World: Part I  ...          NaN  11000000
18  296   History of the World: Part I  ...          NaN  11000000
19  320                     Cat People  ...          NaN  18000000
20  321                     Cat People  ...          NaN  18000000
21  323            Conan the Barbarian  ...   37567440.0  20000000
22  324            Conan the Barbarian  ...   37567440.0  20000000
23  325                      Creepshow  ...          NaN   8000000
24  326                      Creepshow  ...          NaN   8000000
25  331                    First Blood  ...          NaN  14000000
26  332                    First Blood  ...          NaN  14000000
27  338                    Poltergeist  ...   76600000.0  10700000
28  339                    Poltergeist  ...   76600000.0  10700000
29  370      A Nightmare on Elm Street  ...   26505000.0   1800000
30  371      A Nightmare on Elm Street  ...   26505000.0   1800000
31  380                      Footloose  ...   80000000.0   8200000
32  381                      Footloose  ...   80000000.0   8200000
33  396                 The Karate Kid  ...   90800000.0   8000000
34  397                 The Karate Kid  ...   90800000.0   8000000
35  407                Day of the Dead  ...          NaN   3500000
36  408                Day of the Dead  ...          NaN   3500000
37  426  The Return of the Living Dead  ...          NaN   4000000
38  427  The Return of the Living Dead  ...          NaN   4000000
39  497             Dangerous Liaisons  ...   34700000.0  14000000
40  498             Dangerous Liaisons  ...   34700000.0  14000000
41  529                          Glory  ...   26830000.0  18000000
42  530                          Glory  ...   26830000.0  18000000
43  578                   Total Recall  ...  119412921.0  65000000
44  579                   Total Recall  ...  119412921.0  65000000
45  850                         Hamlet  ...    4414535.0  18000000
46  851                         Hamlet  ...    4414535.0  18000000

[47 rows x 9 columns]

When filtering our data to movies before 1997, we find out that movies with their duplicate amount to 47. Therefore, when performing our analysis, we will have to take into account two filters/conditions when constructing our queries:

- Movies released before 1997
- Retrieving movies and not including their duplicates (given we cannot modify the database schema by dropping rows )

### 2. There seems to be a lot of missing data in the gross and budget columns. How many rows have missing data? What would you recommend your manager to do with these rows?

There is a lot of missing data in the gross and budget columns.....but is it only these columns that contain missing data though? to accurately respond to this question, we will query our data for two answers :

- How many rows have missing data in the table?
- How many rows have missing data in the gross and budget columns?

In [36]:
-- Querying for rows with missing data in the entire table
SELECT COUNT(*)
FROM cinema.films
WHERE id IS NULL
   OR title IS NULL
   OR release_year IS NULL
   OR country IS NULL
   OR duration IS NULL
   OR language IS NULL
   OR certification IS NULL
   OR gross IS NULL
   OR budget IS NULL;

count
0   1077

In [37]:
-- Querying for rows with missing data in the gross and budget columns
-- Select the count of rows with missing gross or budget values
SELECT COUNT(id) as rows_with_missing_gross_or_budget
FROM cinema.films
-- Filter rows where either gross or budget is null
WHERE gross IS null OR budget IS null;

rows_with_missing_gross_or_budget
0                               1076

**Answer** : We have 1,077 rows with missing data in the table, but 1,076 rows with missing data in the gross and budget columns combined. Let us keep in mind that this count takes into account duplicates as well. 

   Let us further our analysis by verifying the count of rows with missing data from a particular column only. If only a certain column shows a significant count of missing data, then we know how and where to focus our attention for addressing missing data.
  

In [38]:
-- Select the count of rows where the 'gross' column is missing (null)
SELECT COUNT(id) as rows_with_missing_gross
FROM cinema.films
WHERE gross IS null;

rows_with_missing_gross
0                      810

In [39]:
-- Select the count of rows where the budget column is null
SELECT COUNT(id) as rows_with_missing_budget
FROM cinema.films
WHERE budget IS null;

rows_with_missing_budget
0                       430

In [40]:
-- Select the count of rows where both 'gross' and 'budget' columns are NULL
SELECT COUNT(id) as rows_with_missing_gross_and_budget
FROM cinema.films
WHERE gross IS null AND budget IS null;

rows_with_missing_gross_and_budget
0                                 164

In summary, we find there are 810 rows without a gross value, 430 rows without a budget value, and 164 rows missing both a budget and gross value. In total, there are 1,076 rows (810 + 430 - 164).

These are significant numbers, with the gross column having the most missing data.

### 3. How many different certifications or ratings are present in the database?

In [41]:
-- Select the count of distinct certification values 
SELECT COUNT(DISTINCT certification) AS Certification_count
FROM cinema.films;

certification_count
0                   13

Our count indicates that there are 13 distinct certifications. However, let us review these certifications to verify the accuracy of our data.

In [42]:
-- Select unique certification values 
SELECT DISTINCT certification
FROM cinema.films;

certification
0        Unrated
1              M
2              G
3          NC-17
4             GP
5             PG
6       Approved
7              X
8          PG-13
9               
10             R
11     Not Rated
12        Passed

One of the entries for certifications is empty. This implies there are actually **12 certifications**.

**Answer**: There are 12 certifications.

### 4. What are the top five countries in terms of number of movies produced?

**General Scope**

In [43]:
-- Select the country and the count of distinct movie titles to avoid duplicates
SELECT country, COUNT(DISTINCT title) AS Movie_Count
-- From the cinema.films table
FROM cinema.films
-- Group the results by country
GROUP BY country
-- Order the results by the movie count in descending order
ORDER BY Movie_Count DESC

country  movie_count
0                  USA         3655
1                   UK          429
2               France          153
3               Canada          121
4              Germany           94
..                 ...          ...
60          Kyrgyzstan            1
61               Libya            1
62               Egypt            1
63  Dominican Republic            1
64            New Line            1

[65 rows x 2 columns]

**Movies Released Before 1997**

In [12]:
-- Select the country and the count of distinct movie titles as Movie_Count
SELECT country, COUNT(DISTINCT title) AS Movie_Count
-- From the table cinema.films
FROM cinema.films
-- Where the release year of the films is greater than 1997
WHERE release_year < 1997
-- Group the results by country
GROUP BY country
-- Order the results by Movie_Count in descending order
ORDER BY Movie_Count DESC

country  movie_count
0            USA          725
1             UK           90
2         France           15
3      Australia           13
4         Canada           12
5          Italy           10
6          Japan            6
7   West Germany            3
8      Hong Kong            2
9    Netherlands            2
10   New Zealand            2
11       Germany            2
12   South Korea            1
13  Soviet Union            1
14        Sweden            1
15   Switzerland            1
16       Denmark            1
17         China            1
18          Iran            1
19        Brazil            1
20         Libya            1
21        Norway            1
22          Peru            1

**Answer**: According to our query, for the scope, the top five countries are **USA, UK, France, Canada, and Germany**.

While for movies released before 1997, the top 5 are **USA, UK, France, Australia and Canada.

Apparently Germany wasn't very involved in film-making during the early 90s

### 5. What is the average duration of English versus French movies? 

Here, we handle duplicates to ensure we have a response as accurate as possible

**General Scope**

In [45]:
-- Select the language and the average duration of films
SELECT language, AVG(duration) AS average_duration
-- From the films table in the cinema schema
FROM cinema.films
-- Filter the results to include only films in English or French
WHERE (language = 'English' AND (title, release_year, duration) NOT IN (
    -- Subquery to find combinations of title, release_year, duration
    -- that appear more than once in the cinema.films table.
    SELECT title, release_year, duration
    FROM cinema.films
    GROUP BY title, release_year, duration
    HAVING COUNT(*) > 1
))OR (language = 'French'
AND (title, release_year, duration) NOT IN (
    -- Subquery to find combinations of title, release_year, duration
    -- that appear more than once in the cinema.films table.
    SELECT title, release_year, duration
    FROM cinema.films
    GROUP BY title, release_year, duration
    HAVING COUNT(*) > 1
))
-- Group the results by language to calculate the average duration for each language
GROUP BY language;

language  average_duration
0   French        104.666667
1  English        107.378015

**Movies Released Before 1997**

In [46]:
-- Select the language and the average duration of films
SELECT language, AVG(duration) AS average_duration
-- From the films table in the cinema schema
FROM cinema.films
-- Filter the results to include only films in English or French
WHERE (language = 'English' AND cinema.films.release_year < 1997 AND (title, release_year, duration) NOT IN (
    -- Subquery to find combinations of title, release_year, duration
    -- that appear more than once in the cinema.films table.
    SELECT title, release_year, duration
    FROM cinema.films
    GROUP BY title, release_year, duration
    HAVING COUNT(*) > 1
))OR (language = 'French'
AND cinema.films.release_year < 1997 AND (title, release_year, duration)  NOT IN (
    -- Subquery to find combinations of title, release_year, duration
    -- that appear more than once in the cinema.films table.
    SELECT title, release_year, duration
    FROM cinema.films
    GROUP BY title, release_year, duration
    HAVING COUNT(*) > 1
))
-- Group the results by language to calculate the average duration for each language
GROUP BY language;

language  average_duration
0   French        108.285714
1  English        115.025060

**Answer**:

**Movies Released Before 1997**

- For English movies: 115 minutes
- For French movies: 108 minutes

**For the Entire Dataset**

- For English movies: 107 minutes
- For French movies: 104 minutes

The average duration of English and French movies is approximately 107 minutes and 104 minutes, respectively, for our entire dataset. In contrast, it is 115 minutes and 108 minutes, respectively, for movies released before 1997. This suggests that the majority of viewers may not prefer excessively lengthy movies, with a duration range of 104 to 115 minutes being most ideal.

## 6. Other Insights

6.1  Which rating do most movies have?

In [47]:
-- Select the certification and count of films for each certification
SELECT certification, COUNT(*) AS rating_count
-- From the films table in the cinema schema
FROM cinema.films
-- Filter the films to include only those released before 1997
WHERE release_year < 1997
-- Group the results by the certification column
GROUP BY cinema.films.certification
-- Order the results by the count of films in descending order
ORDER BY rating_count DESC;

certification  rating_count
0              R           402
1             PG           199
2          PG-13           130
3       Approved            55
4              G            34
5      Not Rated            26
6        Unrated            22
7              X            13
8                           13
9         Passed             9
10            GP             6
11             M             5
12         NC-17             3

The majority of the movies are rated 'R'. Let us verify the number of English movies with this rating.

In [48]:
-- Select the count of films and their language
SELECT COUNT(*), language
-- From the cinema.films table
FROM cinema.films
-- Where the certification is 'R', the language is 'English', and the release year is before 1997
WHERE certification = 'R' AND language = 'English' AND release_year < 1997
-- Group the results by language
GROUP BY language;

count language
0    387  English

About 96% of English movies are rated R

6.2 Let us visualise the profit margin for top movies

In [14]:
-- Select the country, language, gross, budget, and calculated profit for films
SELECT title, country, language, gross, budget, (gross - budget) AS profit
-- From the cinema.films table
FROM cinema.films
-- Only include rows where gross and budget are not null and the release year is before 1997
WHERE gross IS NOT null AND budget IS NOT null AND release_year < 1997 AND id NOT IN (
    -- Subquery to find IDs of films that have duplicate combinations of country, language, gross, budget
    SELECT MIN(id)
    FROM cinema.films
    GROUP BY country, language, gross, budget
    HAVING COUNT(*) > 1
)
-- Order the results by profit in descending order
ORDER BY profit DESC;

title country  ...      budget      profit
0            Star Wars: Episode IV - A New Hope     USA  ...    11000000   449935665
1                    E.T. the Extra-Terrestrial     USA  ...    10500000   424449459
2                                 The Lion King     USA  ...    45000000   377783777
3                                 Jurassic Park     USA  ...    63000000   293784000
4    Star Wars: Episode VI - Return of the Jedi     USA  ...    32500000   276625409
..                                          ...     ...  ...         ...         ...
644                               Les visiteurs  France  ...    50000000   -49300000
645                                 Hudson Hawk     USA  ...    70000000   -52781920
646                                  Waterworld     USA  ...   175000000   -86753780
647                            Cutthroat Island     USA  ...    98000000   -87000000
648                                       Akira   Japan  ...  1100000000 -1099560838

[649 rows x 6 columns]

The top 10 movies with the highest revenues are all in English and produced in the USA. This observation suggests that USA-produced movies may be more appealing and have a larger, more active audience than those from other countries.

Now, Instead of visualizing the amount of profit per movie, let us examine the percentage profit to obtain a more realistic view of their earnings compared to other movies. A large profit could mislead us into thinking a movie is more successful than others, whereas it might be due to a large budget.


In [50]:
SELECT 
    id, 
    title, 
    release_year, 
    country, 
    duration, 
    language, 
    certification, 
    gross, 
    budget, 
    (gross - budget) * 100.0 / budget AS percentage_profit
FROM 
    cinema.films
WHERE 
    release_year < 1997
    AND gross IS NOT NULL
    AND budget IS NOT NULL
	AND (title, release_year, duration) NOT IN (
    -- Subquery to find combinations of title, release_year, duration
    -- that appear more than once in the cinema.films table.
    SELECT title, release_year, duration
    FROM cinema.films
    GROUP BY title, release_year, duration
    HAVING COUNT(*) > 1
)
ORDER BY 
    percentage_profit DESC;

id                            title  ...      budget percentage_profit
0    800            The Brothers McMullen  ...       25000      40886.400000
1    622                      El Mariachi  ...        7000      29056.000000
2    179                American Graffiti  ...      777000      14700.514801
3    207                            Rocky  ...      960000      12112.004896
4     15  Snow White and the Seven Dwarfs  ...     2000000       9146.274250
..   ...                              ...  ...         ...               ...
632  664                         Madadayo  ...    11900000        -99.589445
633  110                     Major Dundee  ...     3800000        -99.608605
634   40           The Lady from Shanghai  ...     2300000        -99.655348
635  273                      Raging Bull  ...    18000000        -99.748611
636  490                            Akira  ...  1100000000        -99.960076

[637 rows x 10 columns]

This observation, together with the previous queries, lead us to acknowledge that not only are the highest revenue-generating movies made in USA, but most of them are in English (with a single Spanish movie inserting itself in the rank of top 10). 

English movies seem to dominate our dataset. Let us verify the count of the different languages to view exactly how many English movies we have

In [51]:
-- Select the language and the count of films for each language
SELECT language, COUNT(*) AS language_count
-- From the films table in the cinema schema
FROM cinema.films
-- Only include films released before the year 1997
WHERE release_year <  1997 AND (title, release_year, duration) NOT IN (
    -- Subquery to find combinations of title, release_year, duration
    -- that appear more than once in the cinema.films table.
    SELECT title, release_year, duration
    FROM cinema.films
    GROUP BY title, release_year, duration
    HAVING COUNT(*) > 1
)
-- Group the results by language
GROUP BY language
-- Order the results by the count of films in descending order
ORDER BY language_count DESC;

language  language_count
0      English             838
1       French               7
2      Italian               5
3                            4
4       German               4
5     Japanese               3
6    Cantonese               2
7         None               1
8      Spanish               1
9      Russian               1
10     Swedish               1
11  Portuguese               1
12      Danish               1
13       Dutch               1

838 English movies in total.....with very few movies in other languages released before 1997. 

**Inference**

After thoroughly examining the dataset, conducting our analysis, and identifying plausible patterns, my recommendations for the next potential blockbuster are as follows: 

The film should be in English, with a duration ranging from 1 hour and 40 minutes to a maximum of 2 hours, and should have an R-rating. To reach a broader audience and considering budget constraints, the producers might consider offering a translated or transcribed version in French, as it is the second most spoken language and promotes inclusivity.